crates/egui/src/input_state/mod.rs

```diff
@@ -605,6 +610,7 @@ impl InputState {
             || self.unprocessed_scroll_delta.abs().max_elem() > 0.2
             || self.unprocessed_scroll_delta_for_zoom.abs() > 0.2
             || !self.events.is_empty()
+            || self.needs_repaint
```

Comment:
I think a more robust solution would be to compute this at the beginning of the frame.

As is, we're likely to commit to making the same mistake again if we write something else similar to `consume_key`